aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -70,6 +72,25 @@ module aptos_framework::transaction_validation {
         });
     }
 
+    /// Called in prologue to optionally hold some amount for special txns (e.g. randomness txns).
+    /// `release_to_balance()` should be invoked in the corresponding epilogue with the same arguments.
+    fun hold_from_balance(gas_payer: address, amount: Option<u64>) {
```

Comment:
nit: this probably should be called "collect_gas_deposit" and "refund_gas_deposit"

Reply:
it's a generic framework now, not necessarily "gas"?

Reply:
well `collect|refund_deposit` then?

Reply:
fixed